Make go templating left and right delimiter customizable.

lalyos opened the pull request:
Lately i tried to generate packer.io files. But packer itself uses go templating.

So you have to replace all `{{` with `{{"{{"}}` and the same goes: `}}` to `{{"}}"}}`. It makes the template really hard tpo read. With the propose env variables sigil could use, let's say triple braces to generate files which use go templating:

I'm able to switch on a section with a variable:
```
SIGIL_LEFT_DELIM={{{ SIGIL_RIGHT_DELIM=}}} sigil -f packer.tmpl atlas=true
```
or switch off:
```
SIGIL_LEFT_DELIM={{{ SIGIL_RIGHT_DELIM=}}} sigil -f packer.tmpl atlas=true
```

content of `packer.tmpl`:
```
{
  "variables": {
    "mock": "{{ env `MOCK` }}",

    "aws_access_key": "{{ env `AWS_ACCESS_KEY` }}",
    "aws_secret_key": "{{ env `AWS_SECRET_ACCESS_KEY` }}"
  },

  {{{ with $atlas }}}
  "post-processors": [
    {
      "type": "atlas",
      "artifact": "sequenceiq/{{ user `atlas_artifact` }}",
      "artifact_type": "{{ build_name }}.image"
    }
  ]
  {{{ end }}}
}
```

progrium commented:
Cool. Can we make it a single variable since it's one thing. You'd never just change one of them. Maybe call it `SIGIL_DELIMS` with a value like `{{{,}}}`.

lalyos commented:
Good point. Just changed it, and added a section to README explaining the reason.

Maybe we can define configuration for separating left and right delims in SIGIL_DELIMS like SIGIL_DELIM_SEPARATOR=, 
;)

progrium commented:
lol

Dominik-K commented:
@progrium Unfortunately, the two env variants made it [into the `0.4.0` release](https://github.com/gliderlabs/sigil/pull/15/commits/02d6c11963cf9b2ef718b9950edefe882ad11140). Took me some time to figure this out.
With the [upcoming `0.4.1`](https://github.com/gliderlabs/sigil/pull/34), the `SIGIL_DELIMS` will be used.
